x/globalfee/ante/fee.go

```diff
@@ -97,3 +101,47 @@ func (mfd FeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
 
 	return next(ctx, tx, simulate)
 }
+
+// ParamStoreKeyMinGasPrices type require coins sorted. getGlobalFee will also return sorted coins (might return 0denom if globalMinGasPrice is 0)
+func (mfd FeeDecorator) getGlobalFee(ctx sdk.Context, feeTx sdk.FeeTx) sdk.Coins {
+	var globalMinGasPrices sdk.DecCoins
+	var err error
+	if mfd.GlobalMinFee.Has(ctx, types.ParamStoreKeyMinGasPrices) {
+		mfd.GlobalMinFee.Get(ctx, types.ParamStoreKeyMinGasPrices, &globalMinGasPrices)
+	}
+	// global fee is empty set, set global fee to 0uatom
+	if len(globalMinGasPrices) == 0 {
+		globalMinGasPrices, err = mfd.DefaultZeroGlobalFee(ctx)
+		if err != nil {
+			panic(err)
```

Comment:
IMHO we should avoid panic in the helper method, the panic should stay in the first layer of the code or use test fail. But it is something we should create an issue to try to handle panics from the code and merge this PR as is

Reply:
I changed the panic to antehandler now